pkg/proc/variables.go

```diff
@@ -102,7 +102,7 @@ type Variable struct {
 	DwarfType godwarf.Type
 	RealType  godwarf.Type
 	Kind      reflect.Kind
-	mem       MemoryReadWriter
+	Mem       MemoryReadWriter
```

Comment:
I'd rather not export this at this point, I would suggest that you fish this out with unsafe and write a test for correctness.

Reply:
Using unsafe to access fields in the short term is also an option. I'll handle it by this way for now.

Reply:
I'm actually thinking that you shouldn't need this at all. The only time this is meaningfully different from the object you get from calling `proc.(*Target).Memory` is for stack variables, but really you shouldn't rely on those to find heap references: for one thing they might not be alive anymore and for another there could be (large) chunks of memory kept alive from references that are not reachable from any local variable that we have access to. Think for example the return value of a function that just returned.

Reply:
The 'locals' variable is necessary for memory references. If the memory referenced by these variables becomes invalid and the corresponding heap memory cannot be found, my approach is to skip that variable. And when retrieving references, I also consider the 'args' as a 'locals' variable to prevent the second case.

Reply:
Judging from what you are using I think using `pkg/dwarf/reader.Variables` and then `BinInfo.Location` is more appropriate, if you want to go the route where you are using DWARF (which I think is the wrong approach).

Reply:
I'm considering using gc bitmap to traverse objects, using field offset combined with dwarf types to locate downstream types in order to avoid the issue of potentially missing some live memory by purely traversing the dwarf.

Reply:
> You can't call Locals, it's too high level for what you are doing, you need to read the registers directly or you will not find some live memory. Also, to support range-over-func I will be making changes to `Locals` that will break it completely for your use case.

Get~